numpy/core/src/multiarray/ctors.c

```diff
@@ -1312,7 +1318,11 @@ _array_from_buffer_3118(PyObject *obj, PyObject **out)
     }
     else {
         descr = PyArray_DescrNewFromType(NPY_STRING);
-        descr->elsize = view->itemsize;
+        if (view->itemsize > INT_MAX) {
+            PyErr_SetString(PyExc_ValueError, "Buffer too large");
+            goto fail;
+        }
+        descr->elsize = (int) view->itemsize;
```

Comment:
This is really frustrating. `PyBuffer.itemsize` is a `ssize_t`, but our `descr.itemsize` is an `int`. It'd be great if we could fix this our end, but that would be an ABI breakage.